ginga/util/iohelper.py

```diff
@@ -18,29 +18,48 @@ def get_fileinfo(filespec, cache_dir='/tmp', download=False):
     """
     Parse a file specification and return information about it.
     """
-    idx = None
-    name_ext = ''
+    # Loads first science extension by default.
+    # This prevents [None] to be loaded instead.
+    idx = ('SCI', 1)
+    name_ext = '[SCI,1]'
 
     # User specified an HDU using bracket notation at end of path?
-    match = re.match(r'^(.+)\[(.+)\]$', filespec)
+
+    # This matches
+    #   image.fits[("NAME", 1)]
+    #   image.fits[('NAME', 1)]
+    #   image.fits['NAME', 1]
+    #   image.fits["NAME", 1]
+    #   image.fits[NAME, 1]
+    #   image.fits[(NAME, 1)]
+    match = re.match(r'^(.+)\[\(?[\'\"]?([A-Za-z]+)[\'\"]?\,([0-9])\)?\]$',
+                     filespec)
```

Comment:
Hmm. I did not consider the possibilities of special characters in EXTNAME. It took me quite a while to get that regex working, so I thought I would not want to "waste" the effort. :ghost: But if it is not practical, I can revert this change. Let me know.